src/colorModels/oklch.ts

```diff
@@ -0,0 +1,55 @@
+import { LchaColor, RgbaColor, InputObject } from "../types";
+import { oklabaToRgba, rgbaToOklaba, roundOklaba } from "./oklab";
+import { clamp, isPresent, round } from "../helpers";
+import { ALPHA_PRECISION } from "../constants";
+
+// https://ajalt.github.io/colormath/api/colormath/com.github.ajalt.colormath.model/-oklch/index.html
+
+export const clampOklcha = (oklcha: LchaColor): LchaColor => ({
+  l: clamp(oklcha.l, 0, 1),
+  c: clamp(oklcha.c, 0, 1),
```

Comment:
What is color space for clamp?

Reply:
There is no color space for clamp. It just checks that value > min and < max and fix it

Reply:
This color clamp is useless.

Ir works for RGB/HSL color space, but didn’t work for device independent color space like Lab/Oklab/LCH/OKLCH.

You can encode any possible color in OKLCH. *There is no max for `c`.*

Honestly, C=1 is something beyond visible colors (since 0.4 is max for green in sRGB).

You must specify color space (`srgb`, `p3` or `rec2020`) for color clamp and use complex binary search algorithm.

Read more:
https://bottosson.github.io/posts/gamutclipping/
https://github.com/evercoder/culori/blob/main/src/clamp.js#L28-L70=
https://www.youtube.com/watch?v=dOsp6u4bIwI

Reply:
I can rework the library to postpone the conversion, but in my opinion, it's always necessary. I mean people use the library to convert color to another color space or modify the color parameters. For all of that, it must be converted.

```js
// example
colord({ l: 50, a: 150, b: 150 }) // detect a color modal and convert the input to a master model 
  .darken(0.5) // modify color params in the master modal 
  .toHsl() // convert master modal to HSL
```

It's too expensive (in terms of library size) to have color modifiers for all color models (for example to have `darken` method supporting any color modal).

Having direct model-to-model conversion algorithms (e.g. LAB -> HSL, instead of LAB -> XYZ -> HSL) is also super-expensive because in that case, the library size will be hundreds of KBs (too many combinations).

That's why I prefer performing all of the operations throw a master modal.

Reply:
What color space do you now in `darken()`? Both RGB and XYZ are bad for these transformations

Reply:
The core bundle of colord includes HEX, RGB, HSL and HSV color models only at the moment. Other color models can be installed as plugins.

Built-in `darken`/`lighten`/`saturate` methods use the HSL color model. I know that it's definitely not the best way to do that, but people expect these methods to be available by default (without any plugins) and moreover most of the users expect them to work the same way as it works in all popular libraries and post-processors like SASS (they all use HSL as well). Including, let's say LAB into the default bundle would make the initial library size twice heavier (1,7 KB -> ~3,2 KB gzipped).

Reply:
Talking about ideas. Why not move all methods to separated functions like `import { darkenInHsl } from 'colord'`.

Right now, I will import `darken` even if I do not use it.

Reply:
That's a kind of API, for example, [color2k](https://github.com/ricokahler/color2k) has.
IMO it's quite inconvenient way to work with color, because it usually requires at least 3 methods to call: input/parse value, modify, and convert. 

It's kinda the same as the usage of date-fns instead of day.js (which is way easier to learn and use).

```js
labToLch(darkenInLab(parseLab({ l: 50, a: 150, b: 150 }), 0.5)) // impossible to understand what's going on here
colord({ l: 50, a: 150, b: 150 }).darken().toLch() // super clear
```

I guess we can use your idea and make `lch` and other plugins add their own versions of `darken` and other methods:

```js
colord({ l: 50, a: 150, b: 150 }).darkenInLab(0.5).toLch()
```